### ISSUE TITLE
Better Code Coverage Tool for Kotlin code

### ISSUE THREAD

**Issue by fmueller** [zalando/zally#167]
Our current code coverage solution only measures Java code. Since we decided to migrate more and more code to Kotlin, we need to find a way that this code is also measured for coverage by tests.

**Comment by netme** [comment]
After Kotlin meetup did some investigations:

- We are generating code coverage reports using JaCoCo tool:
   ```bash
   gradle clean build
   open build/reports/jacoco/test/html/index.html 
   ```
- Codacy doesn't accept that part of the report since the file extensions are different than `java`
- We can use codecov.io for code coverage, there is a Kotlin example codecov/example-kotlin

**Comment by netme** [comment]
We can also try https://codebeat.co/

**Comment by fmueller** [comment]
Do these other platforms support multi-language setup with Java/Kotlin? So, can we compare Code Coverage report of Java and Kotlin files?

**Comment by netme** [comment]
I can create an example PR and check how it works.

**Comment by netme** [comment]
Asked Codacy guys via their support channel. Kotlin is not on their short roadmap, but we can build our own Docker image to support it. Here is an example: https://github.com/codacy/codacy-coffeelint 

The question is if it worth it or not.

**Comment by maxim-tschumak** [comment]
I don't think it has very high priority for us. Closing the issue...

**Comment by felix-mueller-sm** [comment]
Just as an information: Codacy does support Kotlin now. :)

**Comment by maxim-tschumak** [comment]
Thanks for the info @fmueller! re-opening the issue.

**Comment by netme** [comment]
Can confirm, got a mail from Codacy yesterday 🎉

**Comment by netme** [comment]
Unfortunately Codacy does not support JaCoCo test coverage reports for Kotlin, already reached their team, they are currently working on that. Moving the ticket back to TODO meanwhile.

**Comment by vadeg** [comment]
Codecov supports test coverage for Kotlin. Fixed in #1223
